File: app/src/main/java/me/imzack/app/end/injector/module/PlanSearchPresenterModule.java
package me.imzack.app.end.injector.module;

import me.imzack.app.end.view.contract.PlanSearchViewContract;

import dagger.Module;
import dagger.Provides;

@Module
public class PlanSearchPresenterModule {

    private final PlanSearchViewContract mPlanSearchViewContract;

    public PlanSearchPresenterModule(PlanSearchViewContract planSearchViewContract) {
        mPlanSearchViewContract = planSearchViewContract;
    }

    @Provides
    PlanSearchViewContract providePlanSearchViewContract() {
        return mPlanSearchViewContract;
    }
}

File: app/src/main/kotlin/me/imzack/app/end/injector/module/PlanSearchPresenterModule.kt
package me.imzack.app.end.injector.module

import dagger.Module
import dagger.Provides
import me.imzack.app.end.view.contract.PlanSearchViewContract

@Module
class PlanSearchPresenterModule(private val mPlanSearchViewContract: PlanSearchViewContract) {

    @Provides
    fun providePlanSearchViewContract() = mPlanSearchViewContract
}
